cartographer_ros/cartographer_ros/node_main.cc

```diff
@@ -480,7 +477,13 @@ void Node::SpinOccupancyGridThreadForever() {
       continue;
     }
     ::nav_msgs::OccupancyGrid occupancy_grid;
-    BuildOccupancyGrid(trajectory_nodes, options_, &occupancy_grid);
+
+    const auto& submaps_options =
+        options_.map_builder_options.trajectory_builder_2d_options()
+            .submaps_options();
+    BuildOccupancyGrid(
+        trajectory_nodes, options_.map_frame, submaps_options.resolution(),
```

Comment:
I would suggest moving in the other direction: Pass the full options. Then you can also move all of this code into a WriteAssets function that calls 2D/3D/common as appropriate.